### PR TITLE
Updates to enforce stage type population

### PR DESCRIPTION
**Pull request by rfleur01**
### Summary:
Today, only ~40% of stages in the Teletraan database have a stage_type that has been updated/populated to a value besides DEFAULT. This PR includes changes that will enforce the population of stage type values besides DEFAULT

### Test Plan:
**Create stage/clone stage**
1. Add a Stage with a Stage Type of DEFAULT
2. Confirm a tooltip shows up 
image
3. Add a stage with a Stage Type of LATEST
image
4. Confirm that the stage was successfully created 
image

**Update stage**
1. Update a configuration value of a stage with stage type equal to DEFAULT
image
3. Confirm that an error banner shows up
image

**Manual deployment**
1. Navigate to the stage with stage type equal to DEFAULT
image
6. Create a deploy 
image
7. Confirm when the create button is click a banner shows up directing the user to update the stage type being unblocked from deploying 
image
image

**Stage is accessed where stage_type = DEFAULT**
image

**User clicks Stage Type Tooltip**
image
image

### CODE REVIEW

**Comment by ntascii** [comment]
probably we can link to a wiki or a doc about Stage type.

**Review by ntascii — APPROVED** [review]

File: deploy-board/deploy_board/templates/configs/env_config.tmpl [hunk 1]
```diff
@@ -249,16 +249,29 @@
...
+                            </select>
+                            <span class="input-group-btn">
+                                <button class="deployToolTip btn stage-type-btn-default" id="stageTypeBtnId" type="button" data-toggle="tooltip" title="click to see more information">
+                                    <span class="glyphicon glyphicon-question-sign"></span>
+                                </button>
+                            </span>
+                        </div>
+                        <div class="col-xs-12" id="stageTypeHelpInfo" hidden="true">
+                            LATEST - Deploys the latest available build</br>
+                            CANARY - Deploy the build after TESTING</br>
```
Comment (liyaqin1) [inline deploy-board/deploy_board/templates/configs/env_config.tmpl:271]: After TESTING? What is TESTING?

File: deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl [hunk 2]
```diff
@@ -25,6 +26,34 @@
...
+                                </select>
+                                <span class="input-group-btn">
+                                    <button class="deployToolTip btn stage-type-btn-default" id="newStageTypeBtnId" type="button" data-toggle="tooltip" title="click to see more information">
+                                        <span class="glyphicon glyphicon-question-sign"></span>
+                                    </button>
+                                </span>
+                            </div>
+                            <div class="col-xs-12" id="newStageTypeHelpInfo" hidden="true">
+                                LATEST - Deploys the latest available build</br>
+                                CANARY - Deploy the build after TESTING</br>
```
Comment (liyaqin1) [inline deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl:51]: After TESTING? What is TESTING?

File: deploy-board/deploy_board/templates/configs/env_config.tmpl (hunk 1 above)
Comment (rfleur01) [reply deploy-board/deploy_board/templates/configs/env_config.tmpl:271]: let me change this to 'tests' or 'testing' to make it more clear

File: deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl (hunk 2 above)
Comment (rfleur01) [reply deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl:51]: see above

File: deploy-board/deploy_board/templates/configs/env_config.tmpl (hunk 1 above)
Comment (liyaqin1) [reply deploy-board/deploy_board/templates/configs/env_config.tmpl:271]: thank you

File: deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl (hunk 2 above)
Comment (liyaqin1) [reply deploy-board/deploy_board/templates/environs/new_stage_modal.tmpl:51]: thanks

**Review by ntascii — APPROVED** [review]

**Pull request merged**